src/field.rs

```diff
@@ -1,8 +1,30 @@
+use std::ops::{BitAnd, Shr};
 use std::{
     fmt::Debug,
     ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign},
 };
 
+// Trait for primitive integer types used to represent the underlying type for field values
+pub trait Int:
+    Sized
+    + Copy
+    + Debug
+    + Ord
+    + Sub<Output = Self>
+    + Into<u128>
+    + Shr<u32, Output = Self>
+    + BitAnd<u8, Output = Self>
```

Comment:
Doesn't this need to be:

```suggestion
    + BitAnd<Self, Output = Self>
```

Reply:
I guess then you would need to define `const ONE: Self` as well.

You could bound with `From<u8>` and drop both `ZERO` and `ONE`.

I'm mostly just concerned that this won't work when the Field uses a different integer basis.

Reply:
yea I was thinking about it too - don't have a good explanation why I decided against it. It should always be safe to upcast from `u8`.